### PR TITLE
Produce constant variables in cases where a SymNode is created with a constant

### PR DESCRIPTION
**Pull request by voznesenskym**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #100144
* #100328

` AOT_DYNAMIC_SHAPES=1 TORCHDYNAMO_DYNAMIC_SHAPES=1  benchmarks/dynamo/huggingface.py --performance  --training --amp --backend eager --disable-cudagraphs --device cuda --only AllenaiLongformerBase --explain` 

Looks promising!

Goes from:

Dynamo produced 173 graphs covering 2760 ops with 160 graph breaks (14 unique)

To:

Dynamo produced 6 graphs covering 2298 ops with 15 graph breaks (7 unique)

cc @soumith @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: torch/_dynamo/variables/tensor.py [hunk 1]
```diff
@@ -466,6 +467,10 @@ def create(cls, tx, proxy, sym_num, **options):
         if sym_num is None:
             sym_num = get_fake_value(proxy.node, tx)
         proxy.node.meta["example_value"] = sym_num
+
+        if isinstance(sym_num, (sympy.Integer, int)):
+            return ConstantVariable(sym_num)
```
Comment (ezyang) [inline torch/_dynamo/variables/tensor.py:472]: You want to coerce the sympy.Integer to int here, no?

**Comment by ezyang** [comment]
This seems fine as an unblocker, but arguably this is just papering over SymNode coverage problems.

Also, @jbschlosser, this affects the multi dispatch dynamo stuff you wrote, mind reviewing it?

File: torch/_dynamo/variables/tensor.py (hunk 1 above)
Comment (voznesenskym) [reply torch/_dynamo/variables/tensor.py:472]: Potentially, I was kind of hoping to keep it a sympy for cases where we .value it into a new (potentially sympy again) variable after some operation, but experimenting on that led me down a useless rabbit hole, so yes, I think thats probably right!

**Comment by voznesenskym** [comment]
> arguably this is just papering over SymNode coverage problems.

This is exactly what this does, no argument from me here!

**Review by ezyang — APPROVED** [review]

**Comment by voznesenskym** [comment]
@pytorcbot rebase

**Comment by ezyang** [comment]
ci failure is real

image

**Comment by voznesenskym** [comment]
Not sure if failures are real. CI is so flaky right now, going to try to yolo this up again. 

Ran all the failures by hand, and they look fine locally.

Screenshot 2023-05-01 at 2 22 12 PM

Here's hoping!

**Pull request closed without merging**